src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -217,6 +235,40 @@ public OpenIdConnectAuthenticationHandler(HttpClient backchannel)
 
             message.State = Options.StateDataFormat.Protect(properties);
 
+            if (Options.UsePostAuthorizationRequests)
+            {
+                using (var buffer = new MemoryStream())
+                using (var writer = new StreamWriter(buffer))
+                {
+                    var inputs = new StringBuilder();
+                    foreach (var parameter in message.Parameters)
+                    {
+                        var name = Options.HtmlEncoder.HtmlEncode(parameter.Key);
+                        var value = Options.HtmlEncoder.HtmlEncode(parameter.Value);
+
+                        inputs.AppendFormat(CultureInfo.InvariantCulture, InputTagFormat, name, value);
+                    }
+
+                    var issuer = Options.HtmlEncoder.HtmlEncode(message.IssuerAddress);
+
+                    writer.Write(string.Format(CultureInfo.InvariantCulture, AuthorizationFormFormat, issuer, inputs));
```

Comment:
We could, but note that `Encoding.UTF8` always emits a BOM. If we don't want BOMs, we'll have to instantiate a custom `UTF8Encoding` with `encoderShouldEmitUTF8Identifier` set to `false`.